documentation/modules/cruise-control/proc-implementing-rebalance-proposal.adoc

```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-implementing-optimization-proposal-{context}']
+
+= Implementing an optimization proposal
+
+You can implement an xref:con-optimization-proposals-{context}[optimization proposal] if its status is `ProposalReady`.
+Cruise Control will then reassign partitions to brokers and change partition leadership, as summarized in the optimization proposal.
+
+[CAUTION]
+====
+*This is not a dry run.* Before you implement an optimization proposal, you must:
+
+* Refresh the proposal in case it has become out of date.
+* Carefully review the contents of the proposal.
```

Comment:
I've added a link to "Contents of optimization proposals". Good suggestion.

I think the extra conceptual information on when to reject a proposal will have to wait. We can capture it as an issue?

Reply:
Are there one or two main reasons why a user would reject an optimization proposal?

Reply:
There are many possible reasons. The main ones might include a really huge amount of data moving or lots of **Inter**Broker replica movements, but I think a true explanation would be quite in depth. It might be better to get the upstream CC community to improve their wiki with this kind of information. I have a couple of WIP PR on the CC docs so I can add this discussion to them.

Reply:
OK, thanks. I think we should capture this discussion in a new Docs issue. Please add me to your CC PRs!